packages/client/src/components/common/modals/upload-documents-modal.tsx

```diff
@@ -0,0 +1,288 @@
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useUploadDocumentsFromGoogleDrive } from '../../../hooks/use-upload-documents-from-google-drive.js';
+import { useUploadMultipleDocuments } from '../../../hooks/use-upload-multiple-documents.js';
+import { Button } from '../../ui/button.js';
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '../../ui/card.js';
+import { Checkbox } from '../../ui/checkbox.js';
+import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle } from '../../ui/dialog.js';
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from '../../ui/form.js';
+import { Input } from '../../ui/input.js';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '../../ui/tabs.js';
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from '../../ui/tooltip.js';
+
+const fileSizeLimit = 10 * 1024 * 1024; // 10MB
+
+// Document Schema
+export const DOCUMENT_SCHEMA = z
+  .instanceof(File)
+  .refine(
+    file =>
+      [
+        'application/pdf',
+        'image/png',
+        'image/jpeg',
+        'image/jpg',
+        'image/svg+xml',
+        'image/gif',
+      ].includes(file.type),
+    { message: 'Invalid document file type' },
+  )
+  .refine(file => file.size <= fileSizeLimit, {
+    message: 'File size should not exceed 5MB',
+  });
+
+const FormSchema = z
+  .object({
+    googleDriveUrl: z
+      .string()
+      .url({ message: 'Invalid url' })
+      .includes('drive.google.com', { message: 'Invalid url: origin must be "drive.google.com"' })
+      .includes('/folders/', { message: 'Invalid url: is this a folder link?' })
+      .optional(),
+    documents: z.array(DOCUMENT_SCHEMA).optional(),
+    method: z.enum(['local-files', 'google-drive']),
+    isSensitive: z.boolean().optional(),
+  })
+  .superRefine((data, ctx) => {
+    if (
+      data.method === 'local-files' &&
+      (data.documents === undefined || data.documents.length === 0)
+    ) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'At least one document is required',
+      });
+    }
+    if (data.method === 'google-drive' && data.googleDriveUrl === undefined) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'The Google Drive URL is required',
+      });
+    }
+  });
+
+export function UploadDocumentsModal({
+  open,
+  onOpenChange,
+  chargeId,
+  onChange,
+}: {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+  chargeId?: string;
+  onChange?: () => void;
+}) {
+  const [tab, setTab] = useState<z.infer<typeof FormSchema>['method']>('local-files');
+  const { uploading: googleDriveUploading, uploadDocumentsFromGoogleDrive } =
+    useUploadDocumentsFromGoogleDrive();
+  const { uploading: multipleFilesUploading, uploadMultipleDocuments } =
+    useUploadMultipleDocuments();
+
+  const form = useForm<z.infer<typeof FormSchema>>({
```

Comment:
Instead of doing this over and over you can just do it once. What I personally do is something like this:

```ts
const formSchema = z.object({});
const FormSchema = z.infer<typeof FormSchema>;
```

Reply:
Very true :) thanks!